src/qml/CSLOLDialogSettings.qml

```diff
@@ -137,6 +140,31 @@ Dialog {
                     text: qsTr("Enable systray icon")
                     checked: false
                     Layout.fillWidth: true
+
+                    onCheckedChanged: {
+                        if (checked) {
+                            startMinimizedCheck.visible = true;
+                            startMinimizedCheck.enabled = true;
+                            createSystemTrayIcon();
+                            // Update the patcher state immediately after creation
+                            if (systemTrayIcon) {
+                                systemTrayIcon.updatePatcherRunning(cslolTools.state === CSLOLTools.StateRunning);
```

Comment:
```
[windowMaximized](qrc:/CSLOLDialogSettings.qml:151: ReferenceError: CSLOLTools is not defined)
```

Reply:
That's my bad, this was left over from a previous change. This if statement can be removed.